### ISSUE TITLE
Ch 19 docker build

### ISSUE THREAD

**Issue by Stegnerd** [miguelgrinberg/microblog#127]
When running the steps to build for docker I was running into issues building. In particular for typed-ast and cffi. I think there was an update to alpine or one of the libraries that removed something along the way. The only way I was able to get past building was to add these two lines to the docker file

.....
COPY requirements.txt requirements.txt
RUN python -m venv venv
RUN apk add --update python3-dev  gcc build-base <----#1
RUN apk add libffi-dev openssl-dev                            <--- #2
RUN venv/bin/pip install -r requirements.txt
  ##......

I just wanted to bring this to your attention

**Comment by miguelgrinberg** [comment]
The problem is with the `pymysql` package. The solution that you used is not good, because it requires the whole gcc compiler to be installed in your docker image, even though it is only needed during installation.

I believe if you use a slightly older version of pymysql this problem is avoided:

    pip install "pymysql<0.9"

**Comment by Stegnerd** [comment]
I noticed that you didn't have pylint in requirements.txt and I believe part of my issue may be cause by pylint bringing in asteroid which required typed-ast which is not part of the alpine. Here is the issue for reference https://github.com/python/typed_ast/issues/63 . If I commented out pylint and asteroid I no longer needed typed-ast and didn't get my errors. Thank you for the fast response and all the work you do!
